src/api/student.js

```diff
@@ -15,4 +16,26 @@ app.get('/', (req, res) => {
     res.successJson(student);
 });
 
+app.post('/', (req, res) => {
+    let workbook = xlsx.read(req.body.data, {type: 'binary'});
+    let studentJSON = xlsx.utils.sheet_to_json(workbook.Sheets[workbook.SheetNames[0]], {range:1});
+    for (let key in studentJSON) {
+        if (studentJSON.hasOwnProperty(key)) {
+            let s = studentJSON[key];
+            let student = {
+                id: s['Student ID'],
+                name: s['Student'],
+                status: s['Status'],
+                email: s['E-mail'],
+                major: s['Major']
+            };
+            if (StudentStore.isCurrentStudent(student) && StudentStore.isNewStudent(student)) {
```

Comment:
The race would only happen if the excel file contains duplicates that are within 1-2 rows of each other i would imagine. Worst case is that `updateStudent` would fail because `students[id]` is undefined.